connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -455,7 +455,9 @@ private WorkerTask buildWorkerTask(ConnectorConfig connConfig,
                     internalKeyConverter, internalValueConverter);
             OffsetStorageWriter offsetWriter = new OffsetStorageWriter(offsetBackingStore, id.connector(),
                     internalKeyConverter, internalValueConverter);
-            KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(producerProps);
+            Map<String, Object> producerConfig = new HashMap<>(producerProps);
```

Comment:
Ah, this also includes a fix to the issue where you were modifying `producerProps`.

Reply:
Yup, what I had before was causing configs defined at the connector level to be applied to all [source] connectors.